src/main/java/org/sagebionetworks/web/client/Portal.java

```diff
@@ -82,18 +81,17 @@ public void onSuccess() {
               // make sure jsni utils code is available to the client
               ginjector.getSynapseJSNIUtils();
 
+              // initialize feature flag config
               ginjector
                 .getSynapseJavascriptClient()
                 .getFeatureFlagConfig(
                   new AsyncCallback<JSONObjectAdapter>() {
                     @Override
                     public void onSuccess(JSONObjectAdapter config) {
                       ginjector
-                        .getSessionStorage()
-                        .setItem(
-                          WebConstants.PORTAL_FEATURE_FLAG_SESSION_STORAGE_KEY,
-                          config.toString()
-                        );
+                        .getFeatureFlagConfig()
+                        .setJson(config.toString());
+                      continueInit();
```

Comment:
Terrible diff below, but all of the initialization code below (that used to run in parallel) has been moved into continueInit() so that it runs serially.  This is to fix the race condition

Reply:
I'd like to convert these into Futures, and wait for them to process in parallel, but I felt like this would be too high risk to patch the staging site on Thursday/Friday, especially when we have this simple (but less optimal) solution.

Reply:
Is there another ticket for this follow-up work?

Reply:
I'll open one up